src/libraries/System.Private.CoreLib/src/System/AggregateException.cs

```diff
@@ -101,9 +101,7 @@ public AggregateException(string? message, Exception innerException)
         /// <exception cref="System.ArgumentException">An element of <paramref name="innerExceptions"/> is
         /// null.</exception>
         public AggregateException(string? message, IEnumerable<Exception> innerExceptions)
-            // If it's already an IList, pass that along (a defensive copy will be made in the delegated ctor).  If it's null, just pass along
-            // null typed correctly.  Otherwise, create an IList from the enumerable and pass that along.
-            : this(message, innerExceptions as IList<Exception> ?? (innerExceptions == null ? (List<Exception>)null! : new List<Exception>(innerExceptions)))
+            : this(message, innerExceptions == null ? Array.Empty<Exception>() : new List<Exception>(innerExceptions).ToArray(), cloneExceptions: false)
```

Comment:
```suggestion
            : this(message, innerExceptions == null ? Array.Empty<Exception>() : innerExceptions.ToArray(), cloneExceptions: false)
```

Wouldn't this be simpler and more efficient?

Reply:
That's linq extension which I think is not worth duplicating in SPC

Reply:
Sorry, I didn't realize that wasn't `Enumerable.ToArray()`.